src/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureSVN.java

```diff
@@ -255,18 +273,14 @@ private boolean findSourceCodeSVN(HttpMessage originalMessage) throws Exception
 			//svnsourcefileattackmsg.setRequestHeader(this.getBaseMsg().getRequestHeader());
 			sendAndReceive(svnsourcefileattackmsg, false);  //do not follow redirects
 
-			//if we got a 404 or a redirect specifically, then this is NOT a match
-			//note that since we are simply relying on the file existing or not, we 
-			//will not attempt any fuzzy matching. Old school.
-			//this check is necessary, otherwise a recursive scan on nodes in the url path cause lots of false positives.
-			if (	svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.NOT_FOUND &&  //404
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.MOVED_PERMANENTLY && //301
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.FOUND && //302
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.SEE_OTHER	&& //303
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.NOT_MODIFIED && //304
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.USE_PROXY && //305 (306 is currently unused)
-					svnsourcefileattackmsg.getResponseHeader().getStatusCode() !=  HttpStatusCode.TEMPORARY_REDIRECT //307
-					) {
+			int attackmsgResponseStatusCode = svnsourcefileattackmsg.getResponseHeader().getStatusCode();
+			//At LOW and MEDIUM ignore all client and server error responses
+			if ((attackStrength==AttackStrength.LOW || attackStrength==AttackStrength.MEDIUM) && 
+				(HttpStatusCode.isClientError(attackmsgResponseStatusCode) || HttpStatusCode.isServerError(attackmsgResponseStatusCode))) {
+					return false; 
+			}				
+			if (!unwantedCodes.contains(attackmsgResponseStatusCode)) {//If the response is wanted (not on the unwanted list)
+
```

Comment:
Shouldn't we be using the alert threshold instead of attack strength?
From the definition of attack strength, higher levels should lead to more number of attacks not (directly to) higher/lower number of false positives. Am I misinterpreting it?

Anyway, in this case continuing with the check will most likely lead to false positives (for example, the ones in the issue report, 400/500 and others likely to happen 401/403, maybe even a 502 or 504). There might be an issue in the server or the connection (or the client ;) but it's not necessarily a SVN source leak.
As an example:
 - Add the following HTTP sender script:
   ```JavaScript
    function sendingRequest(msg, initiator, helper) {
    }

    function responseReceived(msg, initiator, helper) {
        if (initiator != 2)
            return
        msg.setResponseHeader(new org.parosproxy.paros.network.HttpResponseHeader(msg.getResponseHeader().toString().replaceFirst(".+\r\n", "HTTP/1.1 504 Gateway Timeout\r\n")))
}
```

 - Access a file in your local server:
   `http://example.com/no/svn/repo/here/file.txt`
 - Active scan with SVN scanner (with High Strength);
 - Note the issues raised:
    - Source Code Disclosure - SVN (5)
       - GET: http://example.com/no
       - GET: http://example.com/no/svn
       - GET: http://example.com/no/svn/repo
       - GET: http://example.com/no/svn/repo/here
       - GET: http://example.com/no/svn/repo/here/file.txt

Other info of the first alert:
"The source code for [no] was found at [http://example.com/.svn/text-base/no.svn-base]"